core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -117,6 +118,24 @@ abstract class Chunk[+O] {
     }
   }
 
+  def toByteBuffer[B >: O](implicit ev: B =:= Byte): JByteBuffer = {
+    val _ = ev // Convince scalac that ev is used
+    this match {
+      case c: Chunk.Bytes =>
+        JByteBuffer.wrap(c.values, c.offset, c.length)
+      case c: Chunk.ByteBuffer =>
+        val b = c.buf.asReadOnlyBuffer
+        if (c.offset == 0 && b.position == 0 && c.size == b.limit) b
```

Comment:
I don't know that this actually saves a lot.  This trick is inspired by `scodec.bits.ByteVector`.